dev-tools/ls-apis/README.adoc

```diff
@@ -0,0 +1,210 @@
+:showtitle:
+:toc: left
+:icons: font
+
+= ls-apis: list information about Dropshot/Progenitor-based APIs
+
+This tool attempts to identify Progenitor-based API dependencies between Rust components and show information about them.  The focus is on providing information to inform the online upgrade project.
+
+== Quick start
+
+=== Listing APIs and their consumers
+
+```
+$ cargo xtask ls-apis apis
+...
+Nexus Internal API (client: nexus-client)
+    consumed by: dpd (dendrite/dpd)
+    consumed by: omicron-nexus (omicron/nexus)
+    consumed by: omicron-sled-agent (omicron/sled-agent)
+    consumed by: oximeter-collector (omicron/oximeter/collector)
+    consumed by: propolis-server (propolis/bin/propolis-server)
+...
+```
+
+To see what Cargo dependencies caused the tool to identify an API dependency, use `--show-deps`:
+
+```
+$ cargo xtask ls-apis apis --show-deps
+...
+Dendrite DPD (client: dpd-client)
+    consumed by: ddmd (maghemite/ddmd)
+        via path+file:///home/dap/omicron-merge/out/ls-apis/checkout/maghemite/ddmd#0.1.0
+    consumed by: mgd (maghemite/mgd)
+        via path+file:///home/dap/omicron-merge/out/ls-apis/checkout/maghemite/mg-lower#0.1.0
+        via path+file:///home/dap/omicron-merge/out/ls-apis/checkout/maghemite/mgd#0.1.0
+    consumed by: omicron-nexus (omicron/nexus)
+        via path+file:///home/dap/omicron-merge/nexus#omicron-nexus@0.1.0
+    consumed by: omicron-sled-agent (omicron/sled-agent)
+        via path+file:///home/dap/omicron-merge/sled-agent#omicron-sled-agent@0.1.0
+    consumed by: tfportd (dendrite/tfportd)
+        via path+file:///home/dap/omicron-merge/out/ls-apis/checkout/dendrite/tfportd#0.1.0
+    consumed by: wicketd (omicron/wicketd)
+        via path+file:///home/dap/omicron-merge/wicketd#0.1.0
```

Comment:
Thoughts on replacing `dap` with something generic like `user`?

Reply:
I wouldn't mind but I try to avoid munging output that's presented as actual output from a command.  It does seem pretty safe here, admittedly.

Reply:
I'm fine with either :)